Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.cpp

```diff
@@ -208,6 +211,33 @@ void ViewPlatform::requestExitFullScreen()
 }
 #endif // ENABLE(FULLSCREEN_API)
 
+#if ENABLE(GAMEPAD)
+WebPageProxy* ViewPlatform::platformWebPageProxyForGamepadInput()
+{
+    GUniquePtr<GList> toplevels(wpe_toplevel_list());
+    for (GList* iter = toplevels.get(); iter; iter = g_list_next(iter)) {
+        auto* toplevel = WPE_TOPLEVEL(iter->data);
+        if (wpe_toplevel_get_state(toplevel) != WPE_TOPLEVEL_STATE_ACTIVE)
+            continue;
+
+        WPEView* view = nullptr;
+        wpe_toplevel_foreach_view(toplevel, +[](WPEToplevel* toplevel, WPEView* view, gpointer userData) -> gboolean {
```

Comment:
Would it make sense to comment about the rationale of how we pick the view? I guess it is the one visible and focused? I can imagine situations where you can play and write messages in a different view but I don't think we should support that.

Reply:
Yes, we need to find the active window and then get the visible view that gets the input focus